OverReact_Redux_Documentation.md

```diff
@@ -0,0 +1,260 @@
+
+  See: <https://react-redux.js.org/using-react-redux/connect-mapdispatch#defining-mapdispatchtoprops-as-a-function>
+- #### `mergeProps` 
+    - If specified, defines how the final props for the wrapped component are determined.
+If you do not provide `mergeProps`, the wrapped component receives {...ownProps, ...stateProps, ...dispatchProps}
+by default.
+- #### `areStatesEqual` 
+    - Does a simple `==` check by default.
+- #### `areOwnPropsEqual` 
+    - Does a shallow Map equality check by default.
+- #### `areStatePropsEqual` 
+    - Does a shallow Map equality check by default.
+- #### `areMergedPropsEqual` 
+    - Does a shallow Map equality check by default.
+- #### `context` 
+    - Can be utilized to provide a custom context object created with `createContext`.
+`context` is how you can utilize multiple stores. While supported, this is not recommended. :P
+
+    See: <https://redux.js.org/api/store#a-note-for-flux-users>
+    
+    See: <https://stackoverflow.com/questions/33619775/redux-multiple-stores-why-not>
+
+    __Example:__
+    ```dart
+    Store store1 = new Store<CounterState>(counterStateReducer, initialState: new CounterState(count: 0));
+    Store store2 = new Store<BigCounterState>(bigCounterStateReducer, initialState: new BigCounterState(bigCount: 100));UiFactory<CounterProps> ConnectedCounter = connect<CounterState, CounterProps>(
+          mapStateToProps: (state) => (Counter()..intProp = state.count)
+        )(Counter);
+        UiFactory<CounterProps> ConnectedBigCounter = connect<BigCounterState, CounterProps>(
+          mapStateToProps: (state) => (Counter()..intProp = state.bigCount),
+          context: bigCounterContext,
+        )(Counter);
+        react_dom.render(
+          Dom.div()(
+            (ReduxProvider()..store = store1)(
+              (ReduxProvider()
+                ..store = store2
+                ..context = bigCounterContext
+              )(
+                Dom.div()(
+                  Dom.h3()('ConnectedBigCounter Store2'),
+                  ConnectedBigCounter()(
+                    Dom.h4()('ConnectedCounter Store1'),
+                    ConnectedCounter()(),
+                  ),
+                ),
+              ),
+            ),
+          ), querySelector('#content')
+        );
+    ```
+- `pure` if `true` (default), connect performs several equality checks that are used to avoid unnecessary
+calls to `mapStateToProps`, `mapDispatchToProps`, `mergeProps`, and ultimately to `render`. These include
+`areStatesEqual`, `areOwnPropsEqual`, `areStatePropsEqual`, and `areMergedPropsEqual`.
+While the defaults are probably appropriate 99% of the time, you may wish to override them with custom
+implementations for performance or other reasons.
+- `forwardRef` if `true`, the `ref` prop provided to the connected component will be return the wrapped component.
+
+  For more info see: https://react-redux.js.org/api/connect#connect
+  
+## Using Multiple Stores
```

Comment:
Should we just link to here from the `context` section above - and move the examples, etc into this section?

Reply:
I like that idea, it's really tightly coupled and the example fits better in the multiple store section.